frontend/src/Pages/SignUp.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+const SignUp = () => {
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (data: any) => {
```

Comment:
Though it is an option its not a good practice to use `any` as this is basically implying you don't need typescript's strict type checking. So as practice please put the actual properties and type that you need.